pkg/systemd/hwinfo.jsx

```diff
@@ -65,20 +68,115 @@ const SystemInfo = ({ info }) => (
                 </React.Fragment>
             }
             { info.nproc !== undefined &&
-                <tr>
-                    <th>{ _("CPU") }</th>
-                    <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
-                </tr>
+                <React.Fragment>
+                    <tr>
+                        <th>{ _("CPU") }</th>
+                        <td>{ (info.nproc > 1) ? `${info.nproc}x ${info.cpu_model}` : info.cpu_model }</td>
+                    </tr>
+                    <tr>
+                        <th>{ _("CPU Security") }</th>
+                        <td>{ onSecurityClick !== undefined && <a role="link" onClick={onSecurityClick}> Mitigations </a> }</td>
+                    </tr>
+                </React.Fragment>
             }
         </tbody>
     </table>
 );
 
+class CPUSecurityMitigationsDialog extends React.Component {
+    constructor(props) {
+        super(props);
+        this.close = this.close.bind(this);
+        this.saveAndReboot = this.saveAndReboot.bind(this);
+        this.state = {
+            nosmt: undefined,
+        };
+        this.nosmtRow();
+    }
+
+    close() {
+        if (this.props.onClose)
+            this.props.onClose();
+    }
+
+    nosmtRow() {
+        /* nosmt */
+        let promises = [cockpit.spawn(["lscpu", "--json"]), cockpit.spawn(["cat", "/proc/cmdline"])];
+        Promise.all(promises).then(values => {
+            let threads_per_core = Number(JSON.parse(values[0]).lscpu.filter(value => value.field.indexOf("Thread(s) per core") !== -1)[0].data);
+            let smt_disabled = values[1].indexOf("nosmt") !== -1;
+            if (values[1].indexOf("nosmt=") === -1 && (threads_per_core > 1 || (threads_per_core === 1 && smt_disabled)))
+                this.setState({ nosmt: smt_disabled });
+        });
+    }
+
+    saveAndReboot() {
+        this.setState(state => {
+            let options = [];
+            let promises = [];
+
+            if (state.nosmt)
+                options.push(['set', 'nosmt']);
+            else if (state.nosmt !== undefined)
+                options.push(['remove', 'nosmt']);
+
+            for (let option of options)
+                promises.push(cockpit.script(kernelopt_sh, option, { "superuser": true, "err": "message" }));
+
+            Promise.all(promises)
+                    .then(() => {
+                        cockpit.spawn(["shutdown", "--reboot", "now"], { superuser: true, err: "message" });
+                    })
+                    .catch(() => {
+                        this.setState({ rebooting: false });
+                    });
+        });
```

Comment:
Please also catch errors of kernelopt_sh. Let's at least `console.error()` the result for us to be able to debug this, but ideally in an [Alert](https://rawgit.com/patternfly/patternfly-react/gh-pages/patternfly-3/index.html?selectedKind=patternfly-react%2FWidgets%2FAlert&selectedStory=Alert%20types&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs). Changing grub/BLS involves a lot of moving and customizable parts,  so this will certainly fail *somewhere*.